File: data-structures-and-algorithms-ashrafi/java/src/recursive/problem9/Problem9.java
package recursive.problem9;

public class Problem9 {
    
    public static int sumOfFactorials(int n, int s, int i){
        if (i == n) {
            return s;
        }
        return s + sumOfFactorials(n, s * (i + 1), i + 1);
    }

    public static void main(String[] args) {
        int n = 5;
        int sum = 1;
        int i = 1;

        System.out.printf("1! +...+ %d! = %d", n, sumOfFactorials(n, sum, i));
    }
}

File: data-structures-and-algorithms-ashrafi/python/src/recursive/problem_9/problem_9.py
def sum_of_Factorials(n, s, i):
    if i == n:
        return s
    return s + sum_of_Factorials(n, s * (i + 1), i + 1)


def main():
    n = 5
    s = 1
    i = 1

    print(f"1! +...+ {n}! = {sum_of_Factorials(n, s, i)}")

if __name__ == "__main__":
    main()